src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Log4NetLogging/Log4netWrapper.cs

```diff
@@ -59,32 +59,18 @@ private void RecordLogMessage(object logEvent, Type logEventType, IAgent agent)
             // Older versions of log4net only allow access to a timestamp in local time
             var getTimestampFunc = _getTimestamp ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<DateTime>(logEventType, "TimeStamp");
 
-            Func<object, Exception> getLogExceptionFunc;
-
-            try
-            {
-                getLogExceptionFunc = _getLogException ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<Exception>(logEventType, "ExceptionObject");
-            }
-            catch
+            if (_getLogException == null
+                && !VisibilityBypasser.Instance.TryGeneratePropertyAccessor<Exception>(logEventType, "ExceptionObject", out _getLogException)
+                // Legacy property, mainly used by Sitecore
+                && !VisibilityBypasser.Instance.TryGeneratePropertyAccessor<Exception>(logEventType, "m_thrownException", out _getLogException))
```

Comment:
Even though this is more compact and it reduces nesting, I find this harder to read. It is easier for me to understand if there is only 1 method with an out parameter being called in the conditional.

Reply:
Would you prefer something like this? 
```
            if (_getLogException == null)
            {
                if (!VisibilityBypasser.Instance.TryGeneratePropertyAccessor<Exception>(logEventType, "ExceptionObject", out _getLogException))
                {
                    // Legacy property, mainly used by Sitecore
                    if (!VisibilityBypasser.Instance.TryGeneratePropertyAccessor<Exception>(logEventType, "m_thrownException", out _getLogException))
                    {
                        _getLogException = (x) => null;
                    }
                }
            }
```

Reply:
I think that makes it clearer.